php_phongo.c

```diff
@@ -1355,6 +1356,203 @@ static void php_phongo_free_ssl_opt(mongoc_ssl_opt_t *ssl_opt)
 }
 #endif
 
+/* APM callbacks */
+static void php_phongo_dispatch_handlers(const char *name, zval *z_event)
+{
+#if PHP_VERSION_ID >= 70000
+	zval        *value;
+#else
+	HashPosition pos;
+#endif
+	TSRMLS_FETCH();
+
+#if PHP_VERSION_ID >= 70000
+	ZEND_HASH_FOREACH_VAL(&MONGODB_G(subscribers), value) {
+		/* We can't use the zend_call_method_with_1_params macro here, as it
+		 * does a sizeof() on the name argument, which does only work with
+		 * constant names, but not with parameterized ones as it does
+		 * "sizeof(char*)" in that case. */
+		zend_call_method(value, NULL, NULL, name, strlen(name), NULL, 1, z_event, NULL TSRMLS_CC);
+	} ZEND_HASH_FOREACH_END();
+#else
+	zend_hash_internal_pointer_reset_ex(&MONGODB_G(subscribers), &pos);
+	for (;; zend_hash_move_forward_ex(&MONGODB_G(subscribers), &pos)) {
+		zval  **value;
+
+		if (zend_hash_get_current_data_ex(&MONGODB_G(subscribers), (void **) &value, &pos) == FAILURE) {
+			break;
+		}
+
+		/* We can't use the zend_call_method_with_1_params macro here, as it
+		 * does a sizeof() on the name argument, which does only work with
+		 * constant names, but not with parameterized ones as it does
+		 * "sizeof(char*)" in that case. */
+		zend_call_method(value, NULL, NULL, name, strlen(name), NULL, 1, z_event, NULL TSRMLS_CC);
+	}
+#endif
+}
+
+static void php_phongo_command_started(const mongoc_apm_command_started_t *event)
+{
+	mongoc_client_t *client = mongoc_apm_command_started_get_context(event);
```

Comment:
Not needed as far as I know. Our compile flags also create a warning if we do this wrong (through ```-Werror=declaration-after-statement```).

From what I understand, you can define a variable and initialise this, but C90 does not allow to have "just code" before further variable declarations/initialisations.